flutter/flutter
# (insert|move|remove)ChildRenderObject Deprecation: Step 1

## Description

This deprecates the following methods:

* RenderObjectElement.insertChildRenderObject
* RenderObjectElement.moveChildRenderObject
* RenderObjectElement.removeChildRenderObject

... and replaces them with the following methods:

* RenderObjectElement.insertRenderObjectChild
* RenderObjectElement.moveRenderObjectChild
* RenderObjectElement.removeRenderObjectChild

The reason for the deprecation is to provide the `oldSlot` argument to
the `moveRenderObjectChild` method (such an argument was missing from
the now-deprecated `moveChildRenderObject` method) and the `slot`
argument to the `removeRenderObjectChild` method (such an argument was
missing from the now-deprecated `removeChildRenderObject` method). While
no argument was added to `insertRenderObjectChild`, the name change (and
corresponding deprecation) was made to maintain naming parity with the
other two methods.

This initial step does not update or remove any of the `slotToChild`
patterns that exist in the framework. This work is being separated
into two commits in case something needs to be reverted to minimize
the scope of each commit.

## Related Issues

See https://github.com/flutter/flutter/issues/63269 for more info

## Tests

I added the following tests:

* Verify the correct slots are passed to the methods
* Verify that the new methods delegate to the old methods in the base class
* Verify that an assertion error is thrown that informs the developer of the deprecation if old methods are ever invoked

## Checklist


## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].
